### PR TITLE
Add argocdregister controller.

### PR DESCRIPTION
**Pull request by abutcher**
/assign @dgoodwin

### CODE REVIEW

File: pkg/controller/argocdregister/argocdregister_controller.go [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+		return "", fmt.Errorf("secret %s did not contain key %s", secretName, dataKey)
+	}
+	return string(retStr), nil
+}
+
+// getPredictableSecretName generates a unique secret name by hashing the server API URL,
+// which is required as all cluster secrets land in the argocd namespace.
+// This code matches what is presently done in ArgoCD (util/db/cluster.go). However the actual
+// name of the secret does not matter , so we run limited risk of the implementation changing
+// out from underneath us.
```
Comment (dgoodwin) [inline pkg/controller/argocdregister/argocdregister_controller.go:372]: :+1:

**Review by dgoodwin — COMMENTED** [review]
Minor nit, otherwise lgtm.

File: pkg/controller/argocdregister/argocdregister_controller.go [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+	hivemetrics "github.com/openshift/hive/pkg/controller/metrics"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+)
+
+const (
+	ControllerName = "argocdregister"
+
+	argoCDDefaultNamespace   = "argocd"
```
[the remark this answers is not in the document]
Comment (abutcher) [reply pkg/controller/argocdregister/argocdregister_controller.go:61]: Updated!

**Pull request merged**